### DISCUSSION TITLE
Mix up x-transition.delay with Applying CSS classes to get a typewriter effect

### DISCUSSION THREAD

**Discussion by kevinmu17** [alpinejs/alpine#2630] [3. Feature ideas]
Hi there, first of all, what a library! so many smart stuff going around! Listened to you podcast lately with Adam from tailwind, so inspiring!

Okay, so I've found myself in a situation. A client of mine wants some text typed out when the text is in view.
So i've created something like this: (i'm using twig here)
```twig
<span x-show="shown" x-transition.delay.{{ delay }}0ms>{{ character }}</span>
```

Now the above is working fine but I don't quite like the animation, so i've looked over at the docs but can't seem to get it right.

This example isn't working at all. Can't this be mixed?
```twig
<span x-show="shown" x-transition.delay.{{ delay }}0ms x-transition:enter.scale.80 x-transition:leave.scale.90>{{ character }}</span>
```

Ultimately I want to achieve something like:
```twig
<span x-show="shown" x-transition.delay.{{ delay }}00ms x-transition:enter="transition ease-out duration-1000" x-transition:enter-start="opacity-0 scale-90" x-transition:enter-end="opacity-100 scale-100" x-transition:leave="transition ease-in duration-300" x-transition:leave-start="opacity-100 scale-100" x-transition:leave-end="opacity-0 scale-90">{{ character }}</span>
```

[Delay docs](https://alpinejs.dev/directives/transition#customizing-delay)
[Scaling docs](https://alpinejs.dev/directives/transition#customizing-scale)

[Effect to aim for (but a little more tweaked)](https://www.w3schools.com/howto/tryit.asp?filename=tryhow_js_typewriter)

**Comment by SimoTod** [comment]
You can't mix those approaches. Just use the class one (x-transition:enter, etc) and the delay classes from Tailwind.
